extensions-core/druid-basic-security/src/main/java/io/druid/security/basic/authorization/db/updater/CoordinatorBasicAuthorizerMetadataStorageUpdater.java

```diff
@@ -0,0 +1,761 @@
+  public void unassignRole(String prefix, String userName, String roleName)
+  {
+    Preconditions.checkState(lifecycleLock.awaitStarted(1, TimeUnit.MILLISECONDS));
+    unassignRoleInternal(prefix, userName, roleName);
+  }
+
+  @Override
+  public void setPermissions(
+      String prefix, String roleName, List<ResourceAction> permissions
+  )
+  {
+    Preconditions.checkState(lifecycleLock.awaitStarted(1, TimeUnit.MILLISECONDS));
+    setPermissionsInternal(prefix, roleName, permissions);
+  }
+
+  @Override
+  public Map<String, BasicAuthorizerUser> getCachedUserMap(String prefix)
+  {
+    return cachedUserMaps.get(prefix);
+  }
+
+  @Override
+  public Map<String, BasicAuthorizerRole> getCachedRoleMap(String prefix)
+  {
+    return cachedRoleMaps.get(prefix);
+  }
+
+  @Override
+  public byte[] getCurrentUserMapBytes(String prefix)
+  {
+    return connector.lookup(
+        connectorConfig.getConfigTable(),
+        MetadataStorageConnector.CONFIG_TABLE_KEY_COLUMN,
+        MetadataStorageConnector.CONFIG_TABLE_VALUE_COLUMN,
+        getPrefixedKeyColumn(prefix, USERS)
+    );
+  }
+
+  @Override
+  public byte[] getCurrentRoleMapBytes(String prefix)
+  {
+    return connector.lookup(
+        connectorConfig.getConfigTable(),
+        MetadataStorageConnector.CONFIG_TABLE_KEY_COLUMN,
+        MetadataStorageConnector.CONFIG_TABLE_VALUE_COLUMN,
+        getPrefixedKeyColumn(prefix, ROLES)
+    );
+  }
+
+  @Override
+  public Map<String, BasicAuthorizerUser> deserializeUserMap(byte[] userMapBytes)
+  {
+    Map<String, BasicAuthorizerUser> userMap;
+    if (userMapBytes == null) {
+      userMap = Maps.newHashMap();
+    } else {
+      try {
+        userMap = objectMapper.readValue(userMapBytes, USER_MAP_TYPE_REFERENCE);
+      }
+      catch (IOException ioe) {
+        throw new RuntimeException(ioe);
+      }
+    }
+    return userMap;
+  }
+
+  public byte[] serializeUserMap(Map<String, BasicAuthorizerUser> userMap)
+  {
+    try {
+      return objectMapper.writeValueAsBytes(userMap);
+    }
+    catch (IOException ioe) {
+      throw new ISE("WTF? Couldn't serialize userMap!");
+    }
+  }
+
+  @Override
+  public Map<String, BasicAuthorizerRole> deserializeRoleMap(byte[] roleMapBytes)
+  {
+    Map<String, BasicAuthorizerRole> roleMap;
+    if (roleMapBytes == null) {
+      roleMap = Maps.newHashMap();
+    } else {
+      try {
+        roleMap = objectMapper.readValue(roleMapBytes, ROLE_MAP_TYPE_REFERENCE);
+      }
+      catch (IOException ioe) {
+        throw new RuntimeException(ioe);
+      }
+    }
+    return roleMap;
+  }
+
+  public byte[] serializeRoleMap(Map<String, BasicAuthorizerRole> roleMap)
+  {
+    try {
+      return objectMapper.writeValueAsBytes(roleMap);
+    }
+    catch (IOException ioe) {
+      throw new ISE("WTF? Couldn't serialize roleMap!");
+    }
+  }
+
+  private void createUserInternal(String prefix, String userName)
+  {
+    int attempts = 0;
+    while (attempts < numRetries) {
```

Comment:
Maybe backoff is needed between retries?

Reply:
Added backoff